docs/guides/integrating-cert-manager.md

```diff
@@ -0,0 +1,290 @@
+our Gateway resource that we will create in the next step as the solver. To read more about HTTP01 challenges, see
+[here](https://cert-manager.io/docs/configuration/acme/http01/). Use the following YAML definition to create the
+resource, but please note the `email` field should be updated to your own email address.
+
+```yaml
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-prod
+spec:
+  acme:
+    # You must replace this email address with your own.
+    # Let's Encrypt will use this to contact you about expiring
+    # certificates, and issues related to your account.
+    email: my-name@example.com
+    server: https://acme-v02.api.letsencrypt.org/directory
+    privateKeySecretRef:
+      # Secret resource that will be used to store the account's private key.
+      name: issuer-account-key
+    # Add a single challenge solver, HTTP01 using NKG
+    solvers:
+    - http01:
+        gatewayHTTPRoute:
+          parentRefs: # This is the name of the Gateway that will be created in the next step
+            - name: gateway
+              namespace: default
+              kind: Gateway
+```
+
+### Step 3 – Deploy our Gateway with the cert-manager annotation
+
+Next we need to deploy our Gateway. Use can use the below YAML manifest, updating the `spec.linsteners[0].hostname`
+field to the required value for your environment.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+  annotations: # This is the name of the ClusterIssuer created in the previous step
+    cert-manager.io/cluster-issuer: letsencrypt-prod
+spec:
+  gatewayClassName: nginx
+  listeners:
+  - name: http
+    port: 80
+    protocol: HTTP
+  - name: https
+    # Important: The hostname needs to be set to your domain
+    hostname: "cafe.example.com"
+    port: 443
+    protocol: HTTPS
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        name: cafe-secret
+```
+
+It's worth noting a couple of key details in this manifest:
+
+- The cert-manager annotation is present in the metadata – this enables the cert-manager integration, and tells
+- cert-manager which ClusterIssuer configuration it should use for the certificates.
+- There are two Listeners configured – the http listener on port 80 is required for the HTTP01 ACME challenge to work.
+- The hostname needs to set to the required value. A new certificate will be issued from the `letsencrypt-prod`
+  ClusterIssuer for the domain, e.g. "cafe.example.com", once the ACME challenge is successful.
+
+Once the certificate has been issued, cert-manager will create a Certificate resource on the cluster and the
+`cafe-secret` Secret containing the signed keypair in the same Namespace as the Gateway. We can verify the secret has
+been created successfully using `kubectl`. Note it will take a little bit of time for the Challenge to complete and the
+Secret to be created:
+
+```shell
```

Comment:
would it make sense to also add an excerpt of NGINX access logs to show how Let's Encrypt communicated to Cert-Manager through NKG? And perhaps also mention that temp HTTPRoute resource

Reply:
Does the user need to see that level of information? It may add to the complexity - a lot of the point of the integration is the magic that happens behind the scenes.

Reply:
I think it is an important detail that the traffic from Let's Encrypt to Cert-manager will go through NKG. That's why I suggested to update the diagram to reflect that too. 
I think the log can help here - because they will tell the user if Let's Encrypt sent any requests (could be useful for troubleshooting).
But at the same time, reflecting that in the digram will be enough, the logs are nice extras

Reply:
I mentioned the temporary HTTPRoute in the diagram guide (I'm still working on the diagram itself) and again in the  description of the key points of the Gateway yaml. I added an excerpt of the NGINX logs below in the troubleshooting guide. Let me know what you think!